gpt_index/composability/graph.py

```diff
@@ -62,6 +62,8 @@ def _get_default_index_registry() -> IndexRegistry:
     for index_type, index_class in DEFAULT_INDEX_REGISTRY_MAP.items():
         index_registry.type_to_struct[index_type] = index_class.index_struct_cls
         index_registry.type_to_query[index_type] = index_class.get_query_map()
+    # add Node class
+    index_registry.type_to_struct[IndexStructType.NODE] = Node
```

Comment:
@jerryjliu  could you explain what IndexRegistry does again?

Reply:
Yeah the main issue is that the underlying unit that's passed through a composable graph is an index struct (not the outer BaseGPTIndex class). We need a mapping from the index struct type to the index struct class and query class. This is because if you store the graph to disk and reload, you can't save the index struct class or query class mapping (or at least i can't find a way to make it serializable). So we basically need this hardcoded mapping of index type to index struct class and query class, that's passed in during construction and during `load_from_disk`. That's what the index registry is.

Open to make this better or remove it entirely. The high-level idea is thinking about how to "know" what index struct/queries a given struct maps to if it can be serialized/deserialized from disk

Reply:
also this is only needed if we allow for composability, since if we save / load using a given index class, we have info on query classes/index struct class from that index class